src/lang/base/ScillaLexer.mll

```diff
@@ -142,10 +142,11 @@ and read_string buf =
   | eof { raise (Error ("String is not terminated")) }
 
 (* Nested comments, keeping a list of where comments open *)
-and comment depth braces =
+and comment braces =
   parse
-  | "(*"      { comment (depth+1) (lexbuf.lex_curr_p::braces) lexbuf}
-  | "*)"      { if depth = 0 then read lexbuf else comment (depth-1) (tl braces) lexbuf}
-  | newline   { new_line lexbuf; comment depth braces lexbuf}
-  | _         { comment depth braces lexbuf}
-  | eof       { lexbuf.lex_curr_p <- hd braces; raise (Error ("Comment unfinished"))}
+  | "(*"      { comment (lexbuf.lex_curr_p::braces) lexbuf}
+  | "*)"      { if (List.length braces = 1) then read lexbuf
```

Comment:
A couple comments:
- in OCaml parentheses are not necessary in if-expressions before the `then` keyword (`if List.length braces = 1 then ...`);
- `List.length` will traverse the whole list and this is potentially inefficient. In this particular case it might not be a big issue, though. Still, can we replace this line with pattern-matching?
I think that after this change we can merge this PR.